app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableLegacyStockItemIsInStock.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Model\IsProductSalableForRequestedQtyCondition;
+
+use Magento\InventorySales\Model\IsProductSalableCondition\IsSalableLegacyStockItemIsInStock
+    as IsSalableLegacyStockItem;
+use Magento\InventorySalesApi\Api\Data\ProductSalabilityErrorInterfaceFactory;
+use Magento\InventorySalesApi\Api\Data\ProductSalableResultInterfaceFactory;
+use Magento\InventorySalesApi\Api\IsProductSalableForRequestedQtyInterface;
+use Magento\InventorySalesApi\Api\Data\ProductSalableResultInterface;
+
+/**
+ * @inheritdoc
+ */
+class IsSalableLegacyStockItemIsInStock implements IsProductSalableForRequestedQtyInterface
```

Comment:
why do we have this condition here? for IsproductSalableInRequestedQty ? 

Looks like we no need it here